eclipse-scout-core/src/desktop/outline/pages/js/JsPageHelper.ts

```diff
@@ -0,0 +1,470 @@
+        .withElement('page', HybridActionContextElement.of(this.page.outline, this.page))
+    );
+
+    // child pages must be replaced after the first load
+    this._replaceChildPages = true;
+
+    // The LoadChildPagesHybridAction fires the hybridActionEnd-event after the child pages are created and inserted.
+    // The tree and its JsonAdapter may buffer the tree events, but it is ensured that they are contained in the same response from the UI server as the hybridActionEnd-event.
+    // As the session processes all events in the response synchronously the nodesInserted-event is processed before the Promise that is waiting for the hybrid action to complete is resolved.
+    // Therefore, all child pages created on the UI server are already inserted into the tree, and they can be collected into the child pages map.
+    this._addChildPagesToIdMap();
+  }
+
+  /**
+   * Loads the child page for the given id or {@link PageParamDo}.
+   */
+  async loadChildPage(idOrPageParam: string | PageParamDo, replace?: boolean): Promise<Page> {
+    await this.callLoadChildPages(idOrPageParam, replace);
+    return this.findChildPage(idOrPageParam);
+  }
+
+  /**
+   * Loads the child pages for the given ids or {@link PageParamDo}s and returns them in the order of the given ids or {@link PageParamDo}s.
+   */
+  async loadChildPages(idsOrPageParams: string | PageParamDo | (string | PageParamDo)[], replace?: boolean): Promise<Page[]> {
+    await this.callLoadChildPages(idsOrPageParams, replace);
+    return this.findChildPages(idsOrPageParams);
+  }
+
+  /**
+   * Finds the child page for the given id or {@link PageParamDo}.
+   */
+  findChildPage(idOrPageParam: string | PageParamDo): Page {
+    if (!idOrPageParam) {
+      return null;
+    }
+
+    // create id and lookup child page
+    return this._childPagesById.get(this._createChildPageId(idOrPageParam));
+  }
+
+  /**
+   * Finds the child pages for the given ids or {@link PageParamDo}s and returns them in the order of the given ids or {@link PageParamDo}s.
+   */
+  findChildPages(idsOrPageParams: string | PageParamDo | (string | PageParamDo)[]): Page[] {
+    if (!idsOrPageParams) {
+      return [];
+    }
+
+    const idsOrPageParamsArray = arrays.ensure(idsOrPageParams);
+    if (!idsOrPageParamsArray.length) {
+      return [];
+    }
+
+    return idsOrPageParamsArray.map(idsOrPageParam => this.findChildPage(idsOrPageParam));
+  }
+
+  /**
+   * Adds all given child pages to the {@link #_childPagesById}-map. If no child pages are given the child pages of {@link #page} are taken.
+   */
+  protected _addChildPagesToIdMap(childPages?: JsPageChildPage[]) {
+    // use child pages of this.page as default
+    childPages = childPages || this.page.childNodes;
+
+    for (const childPage of childPages) {
+      // extract and assert id
+      const id = this._extractChildPageId(childPage);
+      scout.assertValue(id, `${childPage} does not have a child page param.`);
```

Comment:
This will not be a useful error message, because pages don't have a toString() method. Can we simply skip the page instead?

Same in _removeChildPagesFromIdMap.